packages/contracts/src/dollar/libraries/LibStaking.sol

```diff
@@ -1,532 +1,514 @@
      */
-    function _updateLpPerShare() internal {
-        address stakingShareAddress = LibAppStorage
-            .appStorage()
-            .stakingShareAddress;
-        StakingData storage ss = stakingStorage();
-        StakingShare stake = StakingShare(stakingShareAddress);
-        uint256 lpBalance = IERC20(
-            LibAppStorage.appStorage().stableSwapMetaPoolAddress
-        ).balanceOf(address(this));
-        // the excess LP is the current balance
-        // minus the total deposited LP + LP that needs to be migrated
-        uint256 totalShares = LibChef.totalShares();
-        if (
-            lpBalance >= (stake.totalLP() + ss.totalLpToMigrate) &&
-            totalShares > 0
-        ) {
-            uint256 currentLpRewards = lpBalance -
-                (stake.totalLP() + ss.totalLpToMigrate);
-
-            // is there new LP rewards to be distributed ?
-            if (currentLpRewards > ss.lpRewards) {
-                // we calculate the new accumulated LP rewards per share
-                ss.accLpRewardPerShare =
-                    ss.accLpRewardPerShare +
-                    (((currentLpRewards - ss.lpRewards) * 1e12) / totalShares);
-
-                // update the staking contract lpRewards
-                ss.lpRewards = currentLpRewards;
-            }
-        }
+    function setStakingRewardToken(address newRewardToken) internal {
+        require(newRewardToken != address(0), "Zero address detected");
+        StakingStorage storage stakingStore = stakingStorage();
+        stakingStore.rewardToken = IERC20Ubiquity(newRewardToken);
     }
 
     /**
-     * @notice Mints a staking share on deposit
-     * @param to Address where to mint a staking share
-     * @param lpAmount Amount of LP tokens
-     * @param shares Amount of shares
-     * @param endBlock Staking share end block
-     * @return Staking share id
+     * @notice Sets start block when staking should be active
+     * @param newStartBlock Block number when staking should be active
      */
-    function _mint(
-        address to,
-        uint256 lpAmount,
-        uint256 shares,
-        uint256 endBlock
-    ) internal returns (uint256) {
-        uint256 _currentShareValue = currentShareValue();
-        require(
-            _currentShareValue != 0,
-            "Staking: share value should not be null"
-        );
-        // set the lp rewards debts so that this staking share only get lp rewards from this day
-        uint256 lpRewardDebt = (shares * stakingStorage().accLpRewardPerShare) /
-            1e12;
-        return
-            StakingShare(LibAppStorage.appStorage().stakingShareAddress).mint(
-                to,
-                lpAmount,
-                lpRewardDebt,
-                endBlock
-            );
+    function setStakingStartBlock(uint256 newStartBlock) internal {
+        require(newStartBlock >= block.number, "Can't start in the past");
+        StakingStorage storage stakingStore = stakingStorage();
```

Comment:
I would emit events everywhere we modify the state for easy integration and improved UX